source/fe/mapping_q_cache.cc

```diff
@@ -249,17 +259,465 @@ MappingQCache<dim, spacedim>::initialize(
 {
   AssertDimension(transformation_function.n_components, spacedim);
 
+  this->initialize(mapping,
+                   tria,
+                   [&](const auto &, const auto &point) {
+                     Point<spacedim> new_point;
+                     for (int c = 0; c < spacedim; ++c)
+                       new_point[c] = transformation_function.value(point, c);
+                     return new_point;
+                   },
+                   function_describes_relative_displacement);
+}
+
+
+
+namespace
+{
+  template <typename VectorType,
+            typename std::enable_if<
+              !std::is_same<VectorType,
+                            LinearAlgebra::distributed::Vector<
+                              typename VectorType::value_type>>::value,
+              VectorType>::type * = nullptr>
+  void
+  copy_locally_owned_data_from(
+    const VectorType &vector,
+    LinearAlgebra::distributed::Vector<typename VectorType::value_type>
+      &vector_ghosted)
+  {
+    for (const auto i : vector_ghosted.locally_owned_elements())
+      vector_ghosted[i] = vector[i];
+  }
+
+  template <typename VectorType,
+            typename std::enable_if<
+              std::is_same<VectorType,
+                           LinearAlgebra::distributed::Vector<
+                             typename VectorType::value_type>>::value,
+              VectorType>::type * = nullptr>
+  void
```

Comment:
I'd prefer to do something like

```cpp
template <typename T>
void
copy_locally_owned_data_from(
  const LA::d::V<T> &vector,
  LA::d::V<T> &v_g)
```
here and

```cpp
template <typename VectorType>
void
copy_locally_owned_data_from(
  const VectorType &vector,
  LA::d::V<typename VectorType::value_type> &v_g)
```
above over using SFINAE. I am pretty sure this would work - here's a silly example of this type of deduction:
```cpp
#include <iostream>
#include <list>
#include <vector>

template <typename T>
void
foo(const std::vector<T> &v,
    const std::vector<T> &u)
{
    std::cout << "foo" << std::endl;
}

template <typename Container>
void
foo(const Container &u,
    const std::vector<typename Container::value_type> &v)
{
    std::cout << "foo2" << std::endl;
}

int main()
{
    std::vector<int> u;
    std::vector<int> v;
    std::list<int> w;

    foo(u, v);
    foo(w, v);
}
// output is 
// foo
// foo2
```

Reply:
I would also prefer this approach. But Windows does't ;) 

My first version looked exactly like your suggestion. But Windows complained that it couldn't decide which one to use for `LA:d:V`...

Reply:
That's unfortunate - such is life with MSVC.